src/lib.rs

```diff
@@ -418,7 +426,20 @@ impl Device {
 
     pub fn create_render_pipeline(&self, desc: &RenderPipelineDescriptor) -> RenderPipeline {
         let vertex_entry_point = CString::new(desc.vertex_stage.entry_point).unwrap();
-        let fragment_entry_point = CString::new(desc.fragment_stage.entry_point).unwrap();
+        let vertex_stage = wgn::PipelineStageDescriptor {
+            module: desc.vertex_stage.module.id,
+            entry_point: vertex_entry_point.as_ptr(),
+        };
+        let (_fragment_entry_point, fragment_stage) = if let Some(fragment_stage) = &desc.fragment_stage {
+            let fragment_entry_point = CString::new(fragment_stage.entry_point).unwrap();
+            let fragment_stage = wgn::PipelineStageDescriptor {
+                module: fragment_stage.module.id,
+                entry_point: fragment_entry_point.as_ptr(),
+            };
+            (fragment_entry_point, Some(fragment_stage))
+        } else {
+            (CString::default(), None)
+        };
```

Comment:
I'm uncomfortable with how not-obviously-safe this is, but I couldn't think of a better way :/